### DISCUSSION TITLE
Hydra Crashes

### DISCUSSION THREAD

**Discussion by guneraykut** [ory/hydra#3303] [Q&A]
Since 2-3 weeks, hydra crashes and restarts. Before the restart, high CPU usage is observed. I also obtained logs that say 

`level=error msg=An error occurred audience=application error=map[message:read tcp xx.xx.xx.xx:42098->xx.xx.xx.xx:5432: i/o timeout] `. The second ip belongs to postgresql server. 

DSN params are: `sslmode=require&max_conns=12&max_idle_conns=6`. I also tried to `max_conn_lifetime=1m` and removed all these and no luck. Before and during the crashes, postgresql's CPU, memory and connection limits are fine.  Hydra version is 1.8.5,  postgresql version is 11. All resources run at Azure. 

Is there any suggestion to solve the problem?

**Comment by vinckr** [comment]
I would recommend to upgrade to the latest Hydra version, that should solve it. 
https://www.ory.sh/docs/hydra/self-hosted/upgrade

**Reply by guneraykut** [reply]
It is production machine. Due to service interruption, we planned the upgrade but not near time.
